src/android/BrotherPrinter.java

```diff
@@ -387,6 +394,13 @@ private void setPrinter(JSONArray args, final CallbackContext callbackctx) {
             editor.putString("orientation", printer.orientation != null ? printer.orientation : PrinterInfo.Orientation.LANDSCAPE.toString());
             editor.putString("numberOfCopies", printer.numberOfCopies);
 
+            if(!printer.customPaperFilePath.isEmpty()) {
+                String targetBinFolder = cordova.getActivity()
+                        .getExternalFilesDir("customPaperFileSet/").toString();
+                copyBinFile("www/" + printer.customPaperFilePath, targetBinFolder);
```

Comment:
First @TristanGodal , thank you very much for implementing this.
Most appreciated.

Every time we are setting the printer, we are copying the bin file over and over again.
Not sure this is the optimal way of doing this,

Would it be possible to have this bin file copied only once during installation of the app on the device?

Reply:
> 

Reading the printer manual at 
https://cdn.cnetcontent.com/c5/22/c522c05c-c1d1-493c-8e2f-9fbb02bb2cf5.pdf
I now understand what those bin files are for ... they contain setting commands for the printer.
So, it should be OK to send the command over and over again

Reply:
Sending the paper command (BIN file) to the printer repeatedly is not the concern. SDK requires it, though the printer will persist it and doesn’t need it again as long as nothing else changed it outside the app.

i think you raised a valid point earlier though that the file is being copied every time. I agree that it would be better to copy the file from assets only once (maybe in the APP code and not the plugin…But it may be hard to communicate that to a new plugin user.)